tests/lib/rules/no-confusing-arrow.js

```diff
@@ -71,6 +80,18 @@ ruleTester.run("no-confusing-arrow", rule, {
             code: "var x = (a) => 1 ? 2 : 3",
             output: "var x = (a) => (1 ? 2 : 3)",
             errors: [{ messageId: "confusing" }]
+        },
+        {
+            code: "var x = (a) => 1 ? 2 : 3",
+            output: null,
+            options: [{ onlyOneSimpleParam: true, allowParens: false }],
+            errors: [{ messageId: "confusing" }]
+        },
+        {
+            code: "var x = (a) => 1 ? 2 : 3",
+            output: "var x = (a) => (1 ? 2 : 3)",
+            options: [{ onlyOneSimpleParam: true, allowParens: true }],
+            errors: [{ messageId: "confusing" }]
```

Comment:
Can you also add invalid tests that will check default value for the option, and confirm that the rule doesn't change behavior when the option is not set to `true`:

```js
{
    code: "var x = () => 1 ? 2 : 3",
    output: "var x = () => (1 ? 2 : 3)",
    errors: [{ messageId: "confusing" }]
},
{
    code: "var x = () => 1 ? 2 : 3",
    output: "var x = () => (1 ? 2 : 3)",
    options: [{}],
    errors: [{ messageId: "confusing" }]
},
{
    code: "var x = () => 1 ? 2 : 3",
    output: "var x = () => (1 ? 2 : 3)",
    options: [{ onlyOneSimpleParam: false }],
    errors: [{ messageId: "confusing" }]
},
{
    code: "var x = (a, b) => 1 ? 2 : 3",
    output: "var x = (a, b) => (1 ? 2 : 3)",
    errors: [{ messageId: "confusing" }]
},
{
    code: "var x = (a = b) => 1 ? 2 : 3",
    output: "var x = (a = b) => (1 ? 2 : 3)",
    errors: [{ messageId: "confusing" }]
},
{
    code: "var x = ({ a }) => 1 ? 2 : 3",
    output: "var x = ({ a }) => (1 ? 2 : 3)",
    errors: [{ messageId: "confusing" }]
},
{
    code: "var x = ([a]) => 1 ? 2 : 3",
    output: "var x = ([a]) => (1 ? 2 : 3)",
    errors: [{ messageId: "confusing" }]
},
{
    code: "var x = (...a) => 1 ? 2 : 3",
    output: "var x = (...a) => (1 ? 2 : 3)",
    errors: [{ messageId: "confusing" }]
}
```